serial_communication/web/app.py

```diff
@@ -37,20 +53,27 @@ def index():
     # Baud rates to be displayed in the dropdown
     baud_rates = [9600, 115200, 230400, 460800, 921600]
 
-    return render_template('index.html', default_port=default_serial_port, available_ports=available_ports, baud_rates=baud_rates)
+    return render_template('index.html', default_port=ser, available_ports=available_ports, baud_rates=baud_rates)
 
 @app.route('/read_serial')
 def read_serial():
-    if ser:
-        try:
-            data = ser.readline().decode('utf-8')
-            read_serial_data(data)
-
-            return {'data': data.strip()}  # Strip whitespace from the data
-        except UnicodeDecodeError:
-            return {'error': 'Error decoding serial data'}
-    else:
-        return {'error': 'Serial port not accessible'}
+    global bg_tasks
+    try:
+        if ser:
+            try:
+                data = ser.readline().decode('utf-8')
+                read_serial_data(data)
+
+                return {'data': data.strip()}  # Strip whitespace from the data
+            except UnicodeDecodeError:
+                return {'error': 'Error decoding serial data'}
+        else:
+            return {'error': 'Serial port not accessible'}
+    except serial.SerialException as e:
+        print(f"#1: Error reading from serial port: {e}")
+        bg_tasks = False
+        update_serial_port()
+        return {'error': 'Error reading from serial port'}
```

Comment:
In future please apply this approach on serial_handler file to avoid any issue.

Reply:
Yes I planned i was wondering if i use try catch or also use other logging concept.